src/ring_theory/class_group.lean

```diff
@@ -99,6 +112,35 @@ noncomputable def class_group.mk : (fractional_ideal R⁰ K)ˣ →* class_group
 (quotient_group.mk' (to_principal_ideal R (fraction_ring R)).range).comp
   (units.map (fractional_ideal.canonical_equiv R⁰ K (fraction_ring R)))
 
+lemma class_group.mk_eq_mk {I J : (fractional_ideal R⁰ $ fraction_ring R)ˣ} :
+  class_group.mk I = class_group.mk J
+    ↔ ∃ x : (fraction_ring R)ˣ, I * to_principal_ideal R (fraction_ring R) x = J :=
+begin
+  erw [quotient_group.mk'_eq_mk', canonical_equiv_self, units.map_id],
+  exact ⟨λ ⟨_, ⟨x, hx⟩, h⟩, ⟨x, hx.symm ▸ h⟩, λ ⟨_, h⟩, ⟨_, ⟨_, rfl⟩, h⟩⟩
+end
+
+lemma class_group.mk_eq_mk_of_coe_ideal {I J : (fractional_ideal R⁰ $ fraction_ring R)ˣ}
+  {I' J' : ideal R} (hI : (I : fractional_ideal R⁰ $ fraction_ring R) = I')
+  (hJ : (J : fractional_ideal R⁰ $ fraction_ring R) = J') :
+  class_group.mk I = class_group.mk J
+    ↔ ∃ (x y : R) (hx : x ≠ 0) (hy : y ≠ 0), ideal.span {x} * I' = ideal.span {y} * J' :=
```

Comment:
Thanks! This looks better to me:
```suggestion
lemma class_group.mk_eq_mk_of_coe_ideal
  {I J : (fractional_ideal R⁰ $ fraction_ring R)ˣ} {I' J' : ideal R}
  (hI : (I : fractional_ideal R⁰ $ fraction_ring R) = I')
  (hJ : (J : fractional_ideal R⁰ $ fraction_ring R) = J') :
  class_group.mk I = class_group.mk J
    ↔ ∃ x y : R, x ≠ 0 ∧ y ≠ 0 ∧ ideal.span {x} * I' = ideal.span {y} * J' :=
```
Can you deduce
```
  class_group.mk I = 1 ↔ ∃ x : R, x ≠ 0 ∧ I' = ideal.span {x} :=
```
which will be useful for showing injectivity? That's similar to [`class_group.mk_eq_one_iff`](https://leanprover-community.github.io/mathlib_docs/ring_theory/class_group.html#class_group.mk_eq_one_iff) but here `I` comes from an ideal `I` of `R`.

Reply:
I'm actually doing this as you speak! I should be done within the next hour or so.

Reply:
The proof I have in mind is that if we have `ideal.span {x} * I' = ideal.span {y}`, since `y` is in RHS it's also in LHS so can be expressed as `x` times some element of `I'`, in particular it's divisible by `x`. Now write `y = x * z` and "cancel" `ideal.span {x}` from both sides: one way to do this is to use [submodule.map_injective_of_injective](https://leanprover-community.github.io/mathlib_docs/linear_algebra/basic.html#submodule.map_injective_of_injective), but the lemma `ideal.span {x} * I' = submodule.map (linear_map.lsmul R _ x) I'` seems to be missing.

Reply:
I have an ad-hoc solution that doesn't involve using `submodule.map_injective_of_injective` - let me know what you think!